Fix for -Werror=nested-externs

--- origsrc/intltool-0.50.2/intltool.m4	2012-02-26 12:31:06.000000000 -0600
+++ src/intltool-0.50.2/intltool.m4	2013-04-15 01:44:37.218423800 -0500
@@ -159,9 +159,8 @@ AC_SUBST(ALL_LINGUAS)
 # (copied from glib-gettext.m4)
 if test -z "$DATADIRNAME"; then
   AC_LINK_IFELSE(
-    [AC_LANG_PROGRAM([[]],
-                     [[extern int _nl_msg_cat_cntr;
-                       return _nl_msg_cat_cntr]])],
+    [AC_LANG_PROGRAM([[extern int _nl_msg_cat_cntr;]],
+                     [[return _nl_msg_cat_cntr]])],
     [DATADIRNAME=share],
     [case $host in
     *-*-solaris*)
